client/client.go

```diff
@@ -384,21 +389,7 @@ func (r *NotaryRepository) Publish() error {
 		if err != nil {
 			return err
 		}
-		rootKeyID := r.tufRepo.Root.Signed.Roles["root"].KeyIDs[0]
-		rootKey, ok := r.tufRepo.Root.Signed.Keys[rootKeyID]
-		if !ok {
-			return errors.New(
-				"Root was signed with %s, which is not in its list of keys.")
-		}
-		canonicalRootID, err := tufutils.CanonicalKeyID(rootKey)
-		if err != nil {
-			return err
-		}
-		rootCryptoService, err := r.KeyStoreManager.GetRootCryptoService(canonicalRootID)
-		if err != nil {
-			return err
-		}
-		root, err = r.tufRepo.SignRoot(data.DefaultExpires("root"), rootCryptoService.CryptoService)
+		root, err = r.tufRepo.SignRoot(data.DefaultExpires("root"), r.cryptoService)
```

Comment:
Now that we only have the one cryptoservice, we should be able to remove the final argument from all the SignX functions in tuf.Repo. We only added it in to support unlocked crypto service and I'd love to kill it off again.